### DISCUSSION TITLE
What are the different types of updates ?

### DISCUSSION THREAD

**Discussion by MathRobin** [renovatebot/renovate#15360] [Request Help] [answered]
Hi everybody and thanks to everybody who have contributed to this very great tool! Really, thanks a lot !

I'm a bit confused about update types. For me, semver version is:

major.minor.patch

But with renovate, I discovered pin, digest, rollback and bump. But even if for some of them i think I can understand what they mean, i'm really not sure at all. Maybe I just failed at searching the definition, but i did not succeed to find what are the exact goals of them.

Can anybody explain me please ?
(sorry, my english is probably not really good, i'm french and we have an average poor level in english sadly and i'm not able to improve it more than i would like)

**Comment by HonkingGoose** [comment, marked as answer]
## Major, minor, patch

`major`, `minor` and `patch` are similar to what you expect from SemVer.

`major` = Jest `v27.0.0` -> Jest `v28.0.0`
`minor` = Jest `v27.0.0` -> Jest `v27.1.0`
`patch` = Jest `v27.0.0` -> Jest `v27.0.1`

## Pin

Pins dependencies to a specific version, so like this in a `package.json` file:

```diff
  "dependencies": {
-    "@aws-sdk/client-ec2": "^3.72.0",
  },

  "dependencies": {
+    "@aws-sdk/client-ec2": "3.72.0", <-- The `^` is gone now, and the version is "pinned"
  },
```

I recommend you also read our [deep dive on pinning JavaScript dependencies](https://docs.renovatebot.com/dependency-pinning/). 😉

## Digest

I'll quote from the [docs on Docker digest pinning](https://docs.renovatebot.com/docker/#digest-pinning):

> ## Digest Pinning
> 
> We recommend that you pin your Docker images to an exact digest.
> By pinning to a digest you ensure your Docker builds are **immutable**: every time you do a `pull` you get the same content.
> 
> If you have experience with the way dependency versioning is handled in the JavaScript/npm ecosystem, you might be used to exact versions being immutable.
> e.g. if you specify a version like `2.0.1`, you and your colleagues always get the exact same "code".
> 
> What you may not know is that Docker's tags are not immutable versions, even if they look like a version.
> e.g. you probably expect `myimage:1` and `myimage:1.2` to change over time, but you might incorrectly assume that `myimage:1.2.0` never changes.
> Although it probably _shouldn't_, the reality is that any Docker image tag _can_ change content, and potentially break.
> 
> Using a Docker digest as the image's primary identifier instead of using a Docker tag will achieve immutability.
> It's not easy to work with strings like `FROM node@sha256:d938c1761e3afbae9242848ffbb95b9cc1cb0a24d889f8bd955204d347a7266e`.
> Luckily Renovate can update the digests for you, so you don't have to.
> 
> To keep things simple, Renovate retains the Docker tag in the `FROM` line, e.g. `FROM node:14.15.1@sha256:d938c1761e3afbae9242848ffbb95b9cc1cb0a24d889f8bd955204d347a7266e`.
> Read on to see how Renovate updates Docker digests.

## Rollback

If you use Renovate with `npm` packages, you'll get a rollback PR if the version you're using is deleted from the npm registry.

For example:

1. You're using a JavaScript package called `foo` from npm, on version `1.1.0`
2. Version `1.1.0` gets removed by the npm registry because the maintainer put malware in the package
3. Renovate opens a `rollback` PR to get you back to safety, for example back to version `1.0.0`

[Renovate docs, `rollbackPrs` config option](https://docs.renovatebot.com/configuration-options/#rollbackprs)

## Bump

I'm not sure what `bump` does, I'll let the maintainers explain it. 😉

**Reply by MathRobin** [reply]
Many thanks ! So i have some bugs in my renovate config I have to understand^^

**Comment by norman-zon** [comment]
What does `bump` do?

**Reply by setchy** [reply]
https://docs.renovatebot.com/configuration-options/#bumpversion

**Reply by MathRobin** [reply]
I think, tell me if I'm wrong, it will bump the version of your package.json (in node context) when updating a dependencie. For example, if I ask for automerge patch dependencies, I can ask to auto bump my current package version to a new patch version. For example.

**Reply by jamesbraza** [reply]
From what I can see, `bumpVersion` is a different `matchUpdateTypes`'s `bump`? I am still left wondering what `bump` does

**Reply by rarkins** [reply]
`bumpVersion` is a different concept to `updateType`/`matchUpdateTypes`.

The `bump` updateType is related to rangeStrategy=bump. You can have a patch+bump, minor+bump, or major+bump

**Reply by jamesbraza** [reply]
So basically if you set `matchUpdateTypes` to include `bump`, it means when a version "bump" takes place for a patch version, minor version, or major version, it's considered a match.

I still don't quite understand what is the difference between `matchUpdateTypes": ["minor", "bump"]` and `matchUpdateTypes": ["minor"]`. Like why would `minor` not be eligible for a match, regardless if a version bump is taking place or not?

**Reply by rarkins** [reply]
Please start a new discussion with your questions
